### PR TITLE
Deprecate qml.from_qasm_file

### PR DESCRIPTION
**Pull request by astralcai**
**Context:**
``qml.from_qasm_file`` is an additional function to maintain that does not add much to the already existing ``qml.from_qasm``. It is being deprecated, and users are directed to open the file themselves and load its content using ``qml.from_qasm``.

**Description of the Change:**
Deprecation warning added to ``qml.from_qasm_file``
Example code added to ``qml.from_qasm`` that reads a qasm file and loads with ``qml.from_qasm``

**Related Shortcut Issues:**
[sc-56057]

### CODE REVIEW

File: doc/development/deprecations.rst [hunk 1]
```diff
@@ -9,6 +9,20 @@ deprecations are listed below.
...
+  - Deprecated in v0.36
+  - Will be removed in v0.37
+
+* The ``qml.load`` function is a general-purpose way to convert circuits into PennyLane from other
+  libraries. It is being deprecated in favour of the more specific functions ``from_qiskit``,
+  ``from_qasm``, etc.
+
+  - Deprecated in v0.36
+  - Will be removed in v0.37
+
```
Comment (trbromley) [inline doc/development/deprecations.rst:25]: Wasn't this in a separate PR?

File: doc/development/deprecations.rst [hunk 2]
```diff
@@ -9,6 +9,20 @@ deprecations are listed below.
 Pending deprecations
 --------------------
 
+* ``qml.from_qasm_file`` is an additional function to maintain that does not add much to the already existing
+  ``qml.from_qasm``. It is being deprecated, and users are directed to open the file themselves and load its
```
Comment (trbromley) [inline doc/development/deprecations.rst:13]:
```suggestion
* ``qml.from_qasm_file`` is now deprecated and users are instead directed to open the file
  themselves and load its
```
We don't need to motivate it here.

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -57,6 +58,16 @@
 
 <h3>Deprecations 👋</h3>
 
+* ``qml.load`` is deprecated. Instead, please use the functions outlined in the *Importing workflows* quickstart guide, such as ``qml.from_qiskit``.
+  [(#5312)](https://github.com/PennyLaneAI/pennylane/pull/5312)
+
+* ``qml.from_qasm_file`` is deprecated. Instead, please open the file and then load its content using ``qml.from_qasm``.
+  [(#5331)](https://github.com/PennyLaneAI/pennylane/pull/5331)
+  ```pycon
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:66]:
```suggestion

  ```pycon
```

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -57,6 +58,16 @@
...
+  >>> with open("test.qasm", "r") as f:
+  ...     circuit = qml.from_qasm(f.read())
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:68]: Nice!

File: pennylane/io.py [hunk 5]
```diff
@@ -77,14 +78,24 @@ def load(quantum_circuit_object, format: str, **load_kwargs):
             Currently, only supported for Qiskit's `QuantumCircuit <https://docs.pennylane.ai/projects/qiskit>`_.
 
     Returns:
-        function: the PennyLane template created from the quantum circuit
-        object
+        function: the PennyLane template created from the quantum circuit object
+
+    .. warning::
+        ``qml.load`` is deprecated. Instead, please use the functions outlined in the
+        `Importing workflows <https://docs.pennylane.ai/en/stable/introduction/importing_workflows.html>`_
```
Comment (trbromley) [inline pennylane/io.py:85]: This can be a dynamic link rather than static.

**Review by trbromley — COMMENTED** [review]
Thanks @astralcai!

File: pennylane/io.py (hunk 5 above)
Comment (astralcai) [reply pennylane/io.py:85]: How does dynamic links work?

File: doc/development/deprecations.rst (hunk 1 above)
Comment (astralcai) [reply doc/development/deprecations.rst:25]: This was, but that one hasn't been merged yet, awaiting one more approval. This PR branched off from the other one.

Comment (astralcai) [reply doc/development/deprecations.rst:24]: Oops I didn't realize that when I do that, changes from both PRs are shown in this one. It'll clean up when this one is merged. https://github.com/PennyLaneAI/pennylane/pull/5312/

**Review by albi3ro — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 6]
```diff
@@ -118,6 +118,14 @@
...
   [(#5312)](https://github.com/PennyLaneAI/pennylane/pull/5312)
 
+* ``qml.from_qasm_file`` is deprecated. Instead, please open the file and then load its content using ``qml.from_qasm``.
+  [(#5331)](https://github.com/PennyLaneAI/pennylane/pull/5331)
+
+  ```pycon
+  >>> with open("test.qasm", "r") as f:
+  ...     circuit = qml.from_qasm(f.read())
+  ```
+
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:128]: Please could this also be in the deprecations page?

**Review by Alex-Preciado — APPROVED** [review]

**Pull request merged**